TeamAmaze/AmazeFileManager
# Add support for content:// URI in CompressedExplorerFragment

## Description
As a band-it solution to fix #2650.

When a content:// URI is given, it will query the filename via the ContentResolver, then pass the URI's InputStream to a cache file, which shall be deleted on fragment's destruction or process exit.

Will Toast user if given URI cannot be opened.

There is a downside to this approach. As we will be capturing the whole InputStream for the specified URI, it will become a problem if the content URI's file size is very big.

#### Issue tracker   
Fixes #2650

#### Automatic tests
  
#### Manual tests
  
Device: Pixel 2 emulator
OS: Android 11

Test scenario: open a just downloaded file from Chrome or other web browser, which would trigger providing a content:// URI or downloaded file to CompressedExplorerFragment. It should not crash.

Alternately, try open a compressed file for viewing from Amaze itself. It should not crash either.

#### Build tasks success  
Successfully running following tasks on local:
